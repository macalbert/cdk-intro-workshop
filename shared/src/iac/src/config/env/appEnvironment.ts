export enum AppEnvironment {
    Production = "Production",
    Development = "Development",
    Staging = "Staging",
    GithubAction = "GithubAction",
    Test = "Test",
}
