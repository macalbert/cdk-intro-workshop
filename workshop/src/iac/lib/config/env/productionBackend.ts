import {
    ApiGatewayStack,
    type ApiGatewayStackProps,
    AppEnvironment,
    ApiLambdaStack,
    type ApiLambdaStackProps,
    S3Stack,
    type S3StackProps,
    StackBuildPart,
    type StackBuildProps,
} from "@m47/shared-iac";
import { type Stack } from "aws-cdk-lib";
import { SubnetType } from "aws-cdk-lib/aws-ec2";
import type { IFunction } from "aws-cdk-lib/aws-lambda";

export interface ProductionBackendProps extends StackBuildProps {
    clusterName: string;
    serviceName: string;
    exposedPort: number;
    sourceCodePath: string;
    dockerfileApi: string;
}

export class ProductionBackend extends StackBuildPart {
    props: ProductionBackendProps;

    environment = AppEnvironment.Production;

    constructor(buildProps: ProductionBackendProps) {
        super(buildProps);

        this.props = buildProps;
        this.props.githubRepo = this.formatRepoNameForCloudFormation();
    }

    build(): Stack[] {
        
        const apiLambdaStack = this.createApiLambdaStack(
            "ApiLambda",
            this.props.dockerfileApi,
            "api",
        );

        const apiGatewayStack = this.createApiGateway(
            apiLambdaStack.lambdaFunction,
            "ApiGateway",
            "api-cdk-workshop",
        );

        return [
            apiLambdaStack,
            apiGatewayStack,
        ];
    }

    private createApiLambdaStack(
        name: string,
        dockerfile: string,
        subdomain: string,
    ): ApiLambdaStack {
        const lambdaExportProps: ApiLambdaStackProps = {
            name: name,
            githubRepo: this.props.githubRepo,
            pathDockerFile: this.props.sourceCodePath,
            env: this.props.env,
            vpc: this.props.vpc,
            envName: AppEnvironment.Production,
            timeoutSeconds: 29,
            memorySizeMbs: 1024,
            vpcSubnets: {
                subnetType: SubnetType.PRIVATE_WITH_EGRESS,
            },
            stackName: `${this.props.githubRepo}-${name}`,
            functionName: `${this.props.githubRepo}-${subdomain}`.toLowerCase(),
            dockerFile: dockerfile,
        };

        const stack = new ApiLambdaStack(this.props.scope, lambdaExportProps);

        return stack;
    }

    private createApiGateway(
        lambdaFunction: IFunction,
        name: string,
        subdomain: string,
    ): ApiGatewayStack {
        const apiProps: ApiGatewayStackProps = {
            env: this.props.env,
            name: name,
            envName: AppEnvironment.Production,
            lambdaFunction: lambdaFunction,
            stackName: `${this.props.githubRepo}-${name}`,
            subdomain: subdomain,
            githubRepo: this.props.githubRepo,
            certificateArn: `arn:aws:acm:us-east-1:${this.props.env.account}:certificate/be63062d-5316-47af-9f94-819c1dc02853`,
            domain: "m47.io",
        };

        return new ApiGatewayStack(this.props.scope, apiProps);
    }
}
