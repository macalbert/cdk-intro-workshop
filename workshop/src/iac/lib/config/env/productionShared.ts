import {
    AppEnvironment,
    PipelineStack,
    type PipelineStackProps,
    RdsStack,
    type RdsStackProps,
    StackBuildPart,
    type StackBuildProps,
    WindowsBastionStack,
    type WindowsBastionStackProps,
} from "@m47/shared-iac";
import { RemovalPolicy, type Stack } from "aws-cdk-lib";
import { InstanceType, InstanceClass, InstanceSize } from "aws-cdk-lib/aws-ec2";
import { PostgresEngineVersion } from "aws-cdk-lib/aws-rds";

export class ProductionShared extends StackBuildPart {
    props: StackBuildProps;

    constructor(buildProps: StackBuildProps) {
        super(buildProps);

        this.props = buildProps;
    }

    build(): Stack[] {
        const pipelineStack = this.createPipelineStack();
        const rdsStack = this.createRdsStack();
        const createBastionStack = this.createBastionStack();

        return [
            rdsStack,
            pipelineStack,
            createBastionStack,
        ];
    }

    private createPipelineStack(): PipelineStack {
        const pipeline: PipelineStackProps = {
            name: "Pipeline",
            branch: this.props.branch,
            githubRepo: this.props.githubRepo,
            envName: AppEnvironment.Production,
            secretTokenArn: `arn:aws:secretsmanager:${this.props.env.region}:${this.props.env.account}:secret:github-access-token-secret-dth2kY`,
            testBuildSpec: [
                "workshop/src/iac/buildspecs/test/iac.yml",
                "workshop/src/iac/buildspecs/test/backend.yml",
            ],
            deployBuildSpec: [
                "workshop/src/iac/buildspecs/production/backendApi.yml",
            ],
            testProjectName: `${this.props.githubRepo}-Tests`,
            vpc: this.props.vpc,
            env: this.props.env,
            bucketRemovalPolicy: RemovalPolicy.DESTROY,
            domain: "workshop.com",
            stackName: `${this.formatRepoNameForCloudFormation()}-CodePipeline`,
            manualApproval: false,
            shouldCreateNewSlackChannel: false,
        };

        return new PipelineStack(this.props.scope, pipeline);
    }

    private createRdsStack(): RdsStack {
        const props: RdsStackProps = {
            env: this.props.env,
            vpc: this.props.vpc,
            databaseName: "workshop",
            databaseUserName: "masteruser",
            backupRetentionDays: 14,
            storageSizeGb: 100,
            monitoringIntervalSeconds: 60,
            engineVersion: PostgresEngineVersion.VER_17_4,
            instanceType: InstanceType.of(InstanceClass.T3, InstanceSize.MICRO),
            githubRepo: this.props.githubRepo,
            envName: AppEnvironment.Production,
            stackName: `${this.formatRepoNameForCloudFormation()}-rds`,
            name: "RdsPostgres",
        };

        return new RdsStack(this.props.scope, props);
    }

    private createBastionStack(): WindowsBastionStack {
        const windowsBastionStackProps: WindowsBastionStackProps = {
            vpc: this.props.vpc,
            githubRepo: this.props.githubRepo,
            envName: AppEnvironment.Production,
            name: "bastion",
            stackName: `${this.formatRepoNameForCloudFormation()}-Bastion`,
            env: this.props.env,
        };

        return new WindowsBastionStack(
            this.props.scope,
            windowsBastionStackProps,
        );
    }
}
