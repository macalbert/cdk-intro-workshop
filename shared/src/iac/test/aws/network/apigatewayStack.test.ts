import { App, Stack } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { DockerImageCode, DockerImageFunction } from "aws-cdk-lib/aws-lambda";
import {
	ApiGatewayStack,
	type ApiGatewayStackProps,
} from "../../../src/aws/network/apigatewayStack";
import { AppEnvironment } from "../../../src/config/env/appEnvironment";

describe("ApigatewayStack", () => {
	const env = {
		account: "account",
		region: "eu-west-1",
	};

	test("Should_CreateNewApiGateway_When_StackIsCalled", () => {
		// Arrange
		const stack = new Stack(new App(), "apigatewaystacktest", { env: env });
		const props = createApigatewayStackProps(stack);

		// Act
		const actual = new ApiGatewayStack(stack, props);

		// Assert
		const template = Template.fromStack(actual).toJSON();

		for (const key in template.Resources) {
			const resource = template.Resources[key];

			if (
				typeof key === "string" &&
				key.startsWith("LambdaRestApitestapigatewayDeployment")
			) {
				delete template.Resources[key];
			}
			if (resource.Properties.DeploymentId?.Ref) {
				delete resource.Properties.DeploymentId.Ref;
			}
		}

		expect(template).toMatchSnapshot("apigatewaystacktest");
	});

	function createApigatewayStackProps(app: Stack): ApiGatewayStackProps {
		return {
			env: env,
			subdomain: "shared-api-test",
			name: "GatewayApiStack",
			envName: AppEnvironment.Test,
			githubRepo: "test",
			lambdaFunction: new DockerImageFunction(app, "test-apigateway-function", {
				code: DockerImageCode.fromImageAsset("./test/aws/compute/lambda"),
			}),
			stackName: `test-apigateway`,
			certificateArn: `arn:aws:acm:us-east-1:${env.account}:certificate/someid`,
			domain: "test.com",
		};
	}
});
