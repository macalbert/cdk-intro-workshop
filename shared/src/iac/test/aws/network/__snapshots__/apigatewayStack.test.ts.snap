// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApigatewayStack Should_CreateNewApiGateway_When_StackIsCalled: apigatewaystacktest 1`] = `
{
  "Outputs": {
    "LambdaRestApitestapigatewayEndpointCEA80E3A": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "LambdaRestApitestapigatewayA4C007A6",
            },
            ".execute-api.eu-west-1.",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "LambdaRestApitestapigatewayDeploymentStageprodB8CB301B",
            },
            "/",
          ],
        ],
      },
    },
    "apiUrl": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "LambdaRestApitestapigatewayA4C007A6",
            },
            ".execute-api.eu-west-1.",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "LambdaRestApitestapigatewayDeploymentStageprodB8CB301B",
            },
            "/",
          ],
        ],
      },
    },
    "recordUrl": {
      "Value": "https://shared-api-test.test.com",
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "ARecordtestapigatewayD480FC99": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "LambdaRestApitestapigatewayCustomDomainA5A41848",
              "DistributionDomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::GetAtt": [
              "LambdaRestApitestapigatewayCustomDomainA5A41848",
              "DistributionHostedZoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "shared-api-test.test.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "ApiGatewayAccessLogsFB871B4C": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "RetentionInDays": 365,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "ApiGatewayErrorAlarmtestapigateway59DB36BD": {
      "Properties": {
        "ActionsEnabled": true,
        "AlarmActions": [
          {
            "Ref": "ApiGatewayErrorTopicE8B30FAC",
          },
        ],
        "AlarmDescription": "API Gateway Error Alarm",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": [
          {
            "Name": "ApiName",
            "Value": "LambdaRestApi-test-apigateway",
          },
          {
            "Name": "Stage",
            "Value": {
              "Ref": "LambdaRestApitestapigatewayDeploymentStageprodB8CB301B",
            },
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "5XXError",
        "Namespace": "AWS/ApiGateway",
        "Period": 300,
        "Statistic": "Sum",
        "Threshold": 1,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "ApiGatewayErrorTopicE8B30FAC": {
      "Properties": {
        "DisplayName": "API Gateway Error Topic",
      },
      "Type": "AWS::SNS::Topic",
    },
    "LambdaFunctionErrorAlarmtestapigateway8E385A12": {
      "Properties": {
        "ActionsEnabled": true,
        "AlarmActions": [
          {
            "Ref": "ApiGatewayErrorTopicE8B30FAC",
          },
        ],
        "AlarmDescription": "Lambda Function Error Alarm",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": [
          {
            "Name": "FunctionName",
            "Value": {
              "Fn::ImportValue": "apigatewaystacktest:ExportsOutputReftestapigatewayfunctionBB07AC3DCA481E53",
            },
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "Errors",
        "Namespace": "AWS/Lambda",
        "Period": 300,
        "Statistic": "Sum",
        "Threshold": 1,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "LambdaRestApitestapigatewayA4C007A6": {
      "Properties": {
        "BinaryMediaTypes": [
          "multipart/form-data",
        ],
        "Name": "LambdaRestApi-test-apigateway",
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Project",
            "Value": "test",
          },
          {
            "Key": "StackId",
            "Value": "m47-test-gatewayapistack-test-stack",
          },
        ],
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "LambdaRestApitestapigatewayANYApiPermissionTestapigatewaystacktestm47testgatewayapistackteststackLambdaRestApitestapigateway9CB0C57BANY9D6D55E8": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::ImportValue": "apigatewaystacktest:ExportsOutputFnGetAtttestapigatewayfunctionBB07AC3DArn231706FC",
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:eu-west-1:account:",
              {
                "Ref": "LambdaRestApitestapigatewayA4C007A6",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "LambdaRestApitestapigatewayANYApiPermissionapigatewaystacktestm47testgatewayapistackteststackLambdaRestApitestapigateway9CB0C57BANY0BE7F24E": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::ImportValue": "apigatewaystacktest:ExportsOutputFnGetAtttestapigatewayfunctionBB07AC3DArn231706FC",
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:eu-west-1:account:",
              {
                "Ref": "LambdaRestApitestapigatewayA4C007A6",
              },
              "/",
              {
                "Ref": "LambdaRestApitestapigatewayDeploymentStageprodB8CB301B",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "LambdaRestApitestapigatewayANYDC59B3DA": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:eu-west-1:lambda:path/2015-03-31/functions/",
                {
                  "Fn::ImportValue": "apigatewaystacktest:ExportsOutputFnGetAtttestapigatewayfunctionBB07AC3DArn231706FC",
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Fn::GetAtt": [
            "LambdaRestApitestapigatewayA4C007A6",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "LambdaRestApitestapigatewayA4C007A6",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "LambdaRestApitestapigatewayAccount4AA07316": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "LambdaRestApitestapigatewayA4C007A6",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "LambdaRestApitestapigatewayCloudWatchRoleA99FF14C",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "LambdaRestApitestapigatewayCloudWatchRoleA99FF14C": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Project",
            "Value": "test",
          },
          {
            "Key": "StackId",
            "Value": "m47-test-gatewayapistack-test-stack",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "LambdaRestApitestapigatewayCustomDomainA5A41848": {
      "Properties": {
        "CertificateArn": "arn:aws:acm:us-east-1:account:certificate/someid",
        "DomainName": "shared-api-test.test.com",
        "EndpointConfiguration": {
          "Types": [
            "EDGE",
          ],
        },
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Project",
            "Value": "test",
          },
          {
            "Key": "StackId",
            "Value": "m47-test-gatewayapistack-test-stack",
          },
        ],
      },
      "Type": "AWS::ApiGateway::DomainName",
    },
    "LambdaRestApitestapigatewayCustomDomainMapapigatewaystacktestm47testgatewayapistackteststackLambdaRestApitestapigateway9CB0C57BD5D7FBF6": {
      "Properties": {
        "DomainName": {
          "Ref": "LambdaRestApitestapigatewayCustomDomainA5A41848",
        },
        "RestApiId": {
          "Ref": "LambdaRestApitestapigatewayA4C007A6",
        },
        "Stage": {
          "Ref": "LambdaRestApitestapigatewayDeploymentStageprodB8CB301B",
        },
      },
      "Type": "AWS::ApiGateway::BasePathMapping",
    },
    "LambdaRestApitestapigatewayOPTIONSDD1E4163": {
      "Properties": {
        "ApiKeyRequired": false,
        "AuthorizationType": "NONE",
        "HttpMethod": "OPTIONS",
        "Integration": {
          "IntegrationResponses": [
            {
              "ResponseParameters": {
                "method.response.header.Access-Control-Allow-Credentials": "'true'",
                "method.response.header.Access-Control-Allow-Headers": "'Cache-Control,Content-Language,Content-Type,Expires,Last-Modified,Pragma,Acceptencoding,Authorization,X-Amz-Date,X-Api-Key,X-Forwarded-For'",
                "method.response.header.Access-Control-Allow-Methods": "'OPTIONS,GET,PUT,POST,DELETE,PATCH,HEAD'",
                "method.response.header.Access-Control-Allow-Origin": "'*'",
              },
              "StatusCode": "204",
            },
          ],
          "RequestTemplates": {
            "application/json": "{ statusCode: 200 }",
          },
          "Type": "MOCK",
        },
        "MethodResponses": [
          {
            "ResponseParameters": {
              "method.response.header.Access-Control-Allow-Credentials": true,
              "method.response.header.Access-Control-Allow-Headers": true,
              "method.response.header.Access-Control-Allow-Methods": true,
              "method.response.header.Access-Control-Allow-Origin": true,
            },
            "StatusCode": "204",
          },
        ],
        "ResourceId": {
          "Fn::GetAtt": [
            "LambdaRestApitestapigatewayA4C007A6",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "LambdaRestApitestapigatewayA4C007A6",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "LambdaRestApitestapigatewayproxy278F93EA": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "LambdaRestApitestapigatewayA4C007A6",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": {
          "Ref": "LambdaRestApitestapigatewayA4C007A6",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "LambdaRestApitestapigatewayproxyANYApiPermissionTestapigatewaystacktestm47testgatewayapistackteststackLambdaRestApitestapigateway9CB0C57BANYproxy4077DB5D": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::ImportValue": "apigatewaystacktest:ExportsOutputFnGetAtttestapigatewayfunctionBB07AC3DArn231706FC",
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:eu-west-1:account:",
              {
                "Ref": "LambdaRestApitestapigatewayA4C007A6",
              },
              "/test-invoke-stage/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "LambdaRestApitestapigatewayproxyANYApiPermissionapigatewaystacktestm47testgatewayapistackteststackLambdaRestApitestapigateway9CB0C57BANYproxy8C18D3A8": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::ImportValue": "apigatewaystacktest:ExportsOutputFnGetAtttestapigatewayfunctionBB07AC3DArn231706FC",
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:eu-west-1:account:",
              {
                "Ref": "LambdaRestApitestapigatewayA4C007A6",
              },
              "/",
              {
                "Ref": "LambdaRestApitestapigatewayDeploymentStageprodB8CB301B",
              },
              "/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "LambdaRestApitestapigatewayproxyANYF4B4EAE7": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:eu-west-1:lambda:path/2015-03-31/functions/",
                {
                  "Fn::ImportValue": "apigatewaystacktest:ExportsOutputFnGetAtttestapigatewayfunctionBB07AC3DArn231706FC",
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "LambdaRestApitestapigatewayproxy278F93EA",
        },
        "RestApiId": {
          "Ref": "LambdaRestApitestapigatewayA4C007A6",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "LambdaRestApitestapigatewayproxyOPTIONSC2A4259B": {
      "Properties": {
        "ApiKeyRequired": false,
        "AuthorizationType": "NONE",
        "HttpMethod": "OPTIONS",
        "Integration": {
          "IntegrationResponses": [
            {
              "ResponseParameters": {
                "method.response.header.Access-Control-Allow-Credentials": "'true'",
                "method.response.header.Access-Control-Allow-Headers": "'Cache-Control,Content-Language,Content-Type,Expires,Last-Modified,Pragma,Acceptencoding,Authorization,X-Amz-Date,X-Api-Key,X-Forwarded-For'",
                "method.response.header.Access-Control-Allow-Methods": "'OPTIONS,GET,PUT,POST,DELETE,PATCH,HEAD'",
                "method.response.header.Access-Control-Allow-Origin": "'*'",
              },
              "StatusCode": "204",
            },
          ],
          "RequestTemplates": {
            "application/json": "{ statusCode: 200 }",
          },
          "Type": "MOCK",
        },
        "MethodResponses": [
          {
            "ResponseParameters": {
              "method.response.header.Access-Control-Allow-Credentials": true,
              "method.response.header.Access-Control-Allow-Headers": true,
              "method.response.header.Access-Control-Allow-Methods": true,
              "method.response.header.Access-Control-Allow-Origin": true,
            },
            "StatusCode": "204",
          },
        ],
        "ResourceId": {
          "Ref": "LambdaRestApitestapigatewayproxy278F93EA",
        },
        "RestApiId": {
          "Ref": "LambdaRestApitestapigatewayA4C007A6",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
